rosetta-source/src/main/rosetta/observable-asset-func.rosetta

```diff
@@ -37,3 +37,16 @@ func InterestRateObservableCondition: <"Implementation for PriceQuantity.Interes
                 extract [
                     priceType = PriceTypeEnum -> InterestRate and arithmeticOperator exists
                 ] all = True
+
+func ObservableIsCommodity:  <"Validates whether all of the constituents of an Observable are Commodities.">
+    inputs:
+        observable Observable (0..1)
+    output:
+        valid boolean (1..1)
+
+    set valid:
+        observable switch
+            Asset   then Commodity exists ,
+            Basket  then basketConstituent extract [ Asset -> Commodity exists ] all = True ,
```

Comment:
What if the basket contains commodity indices, shouldn't this loop on itself by calling `ObservableIsCommodity` on `basketConstituent` instead?